src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -264,14 +265,24 @@ public static void UpdateBackgroundLayerFrame(this UIView view)
 
 			var layer = view.Layer;
 
+			UpdateBackgroundLayerFrame(layer, view.Bounds);
+		}
+
+		static void UpdateBackgroundLayerFrame(CALayer layer, CGRect bounds)
+		{
 			if (layer == null || layer.Sublayers == null || layer.Sublayers.Length == 0)
 				return;
 
 			foreach (var sublayer in layer.Sublayers)
 			{
-				if (sublayer.Name == BackgroundLayerName && sublayer.Frame != view.Bounds)
+				UpdateBackgroundLayerFrame(sublayer, bounds);
+			}
+
+			foreach (var sublayer in layer.Sublayers)
```

Comment:
Why are we looping through all the Sublayers twice?

Reply:
You're right. I have applied changes to remove the unnecessary iteration.